src/Swashbuckle.AspNetCore.Annotations/SwaggerParameterAttributeOperationFilter.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.AspNetCore.Mvc.Controllers;
+using Swashbuckle.AspNetCore.Swagger;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Swashbuckle.AspNetCore.Annotations
+{
+    /// <summary>
+    /// An operation filter that adds parameter metadata on an action.
+    /// defined by the <see cref="SwaggerParameterAttribute" /> on controller methods.
+    /// </summary>
+    /// <remarks>
+    /// This filter overrides any existing description and sets the Required option only if set to true, otherwise it leaves it alone.
+    /// </remarks>
+    /// <seealso cref="Swashbuckle.AspNetCore.SwaggerGen.IOperationFilter" />
+    public class SwaggerParameterAttributeOperationFilter : IOperationFilter
+    {
+        public void Apply(Operation operation, OperationFilterContext context)
+        {
+            EnsureParameters(operation);
+
+            foreach (var pair in this.GetParameterAttributes(context))
+            {
+                var param = GetOrAddParameter(operation, pair.Key);
+                param.Description = pair.Value.Description;
+
+                // Only override if required is set to true, meaning that if required is not set it will be whatever the implementation has detected it should be.
+                // This allows for use of description without having to worry about the required parameter as well - though it prevents to option of using it to se a parameter as not required if the implementation thinks it is!
+                param.Required = pair.Value.Required ? true : param.Required;
+            }
+        }
+
+        /// <summary>
+        /// Ensures the parameters exists on the operation
+        /// </summary>
+        /// <param name="operation">The operation.</param>
+        private static void EnsureParameters(Operation operation)
+        {
+            if (operation.Parameters == null)
+            {
+                operation.Parameters = new List<IParameter>();
+            }
+        }
+
+        /// <summary>
+        /// Gets the or add parameter to the operation collection of parameters
+        /// </summary>
+        /// <param name="operation">The operation.</param>
+        /// <param name="name">The name of the parameter</param>
+        /// <returns></returns>
+        private static IParameter GetOrAddParameter(Operation operation, string name)
+        {
+            var parameter = operation.Parameters?.SingleOrDefault(p => p.Name == name);
```

Comment:
Depending on how the action parameter is bound to the request, this may not always locate the parameter correctly. For example, consider the following:

```csharp
public IActionResult GetProducts([FromQuery(Name="kw")]string keywords)
```

In this case, the Swagger `Parameter` will (correctly) have the name `kv` but the name it's being compared to will be `keywords`.

I've had to work around this in the Xml filters and used the following logic to do so:

```csharp
var actionParameter = apiDescription.ActionDescriptor.Parameters
    .FirstOrDefault(p => parameter.Name.Equals(
        (p.BindingInfo?.BinderModelName ?? p.Name), StringComparison.OrdinalIgnoreCase));
if (actionParameter == null) continue;
```

I have a plan to centralize this logic so it doesn't have to be repeated but that requires a broader refactor. For now, I would just repeat it here.

Reply:
Damn, wasnt aware you could override the name in the decoration - will update it later. Should probably also add a separate test for it